src/Bicep.Core/TypeSystem/TypePropertyFlags.cs

```diff
@@ -36,5 +36,10 @@ public enum TypePropertyFlags
         /// This property's value known at the start of the deployment, and does not require inlining.
         /// </summary>
         DeployTimeConstant = 1 << 4,
+
+        /// <summary>
+        /// The property's value must be a known at the start of the deployment when set by users, but it cannot be referenced to set other deploy-time constants. E.g. tags.
+        /// </summary>
+        WeakDeployTimeConstant = 1 << 5,
```

Comment:
The naming of this is a little confusing - would be worth trying to see if we can come up with something better. It feels like (prior to this PR), the `DeployTimeConstant` flag represents two concepts - "the property must be set to a deploy-time value" and "the property can be read at deploy-time"; with this change we are now separating those two concepts out.

I wonder if it's worth breaking the existing flag into two flags to represent those concepts, and assigning both to "id", "name", "type" & "apiVersion". Thoughts?

Reply:
Yeah TBH I'm also not happy with that name, and I like the idea of breaking the existing `DeployTimeConstant` into two flags. I create a new flag called `ReadableAtDeployTime` and assigned it to the standardized properties. I didn't change the name of `DeployTimeConstant` though - let me know if that's fine.